packages/muelu/src/Misc/MueLu_InverseApproximationFactory_def.hpp

```diff
@@ -115,12 +120,83 @@ namespace MueLu {
       const RCP<const Vector> D = (!fixing ? Utilities::GetInverse(diag) : Utilities::GetInverse(diag, 1e-4, one));
       Ainv = MatrixFactory::Build(D);
     }
+    else if(method=="sparseapproxinverse") {
+      Ainv = GetSparseInverse(A, A->getCrsGraph());
+    }
 
     GetOStream(Statistics1) << "Approximate inverse calculated by: " << method << "." << std::endl;
+    GetOStream(Statistics1) << "Ainv has " << Ainv->getGlobalNumRows() << "x" << Ainv->getGlobalNumCols() << " rows and columns." << std::endl;
 
     Set(currentLevel, "Ainv", Ainv);
   }
 
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>>
+  InverseApproximationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::GetSparseInverse(const RCP<Matrix>& Aorg, const RCP<const CrsGraph>& sparsityPattern) const {
+
+    // construct the inverse matrix with the given sparsity pattern
+    RCP<Matrix> Ainv = MatrixFactory::Build(sparsityPattern);
+    Ainv->resumeFill();
```

Comment:
@maxfirmbach Is this one necessary?

Reply:
I know it looks weird, but for Epetra it's necessary ...